src/Upgrade/Infrastructure/PackageManager/CommandExecutor/ComposerCommandExecutor.php

```diff
@@ -288,7 +288,11 @@ protected function getPackageString(PackageCollection $packageCollection): strin
 
                 continue;
             }
-            $package = sprintf('%s:%s', $package->getName(), $package->getVersion());
+            $version = $package->getVersion();
+            if (str_contains($package->getVersion(), ' ')) {
```

Comment:
`str_contains` is a php8 function. Unlucky upgrader has to support 7.4. Use `$version` variable.
Move to separate method, it would be useful for other composer commands in future

Reply:
I guess we are going to remove 7.4)